workflow/scripts/plot_diffexp_heatmap.R

```diff
@@ -10,9 +10,12 @@ library(tidyr)
 sleuth_file <- read.csv(snakemake@input[["logcountmatrix_file"]],
     sep = "\t", header = TRUE
 )
+print(length(snakemake@params[["predef_genelist"]]))
+print(snakemake@params[["predef_genelist"]]$activate)
 # Getting pre-defined genelist file
-if (length(snakemake@input[["predef_genelist"]]) != 0) {
-    predefine_genelist <- read.table(snakemake@input[["predef_genelist"]],
+if (snakemake@params[["predef_genelist"]]$activate == TRUE &&
+    length(snakemake@params[["is_3prime_experiment"]]) == "true") {
```

Comment:
Why do you check for 3prime at all here? A heatmap for a predefined list of genes makes sense in any case.

Reply:
Noted. As stated below I removed the 3prime part. It now checks only if the predefined list is there or not.